pkg/ebpf/tracee.go

```diff
@@ -827,6 +830,54 @@ func (t *Tracee) getFiltersConfig() uint32 {
 	return cFilterVal
 }
 
+// validateKallsymsDependencies load all symbols required by events' dependencies from the kallsyms file to check for missing symbols.
+// If some symbols are missing, it will cancel their event with informative error message.
+func (t *Tracee) validateKallsymsDependencies() {
+	var reqKsyms []string
+	symsToDependentEvents := make(map[string][]events.ID)
+	for id := range t.events {
+		event := events.Definitions.Get(id)
+		for _, symDep := range event.Dependencies.KSymbols {
+			reqKsyms = append(reqKsyms, symDep.Symbol)
+			if symDep.Required {
```

Comment:
I wonder when symbols wouldn't be required (looks like kSymboldependency has required always set to true in all symbols declared).

Reply:
So I have a place I want to do it in already, this is why I added the required flag.
I thought that we can try using symbols that are not guaranteed to exist like symbols in the `data` or `rodata` segments, and have a fail-safe use of `text` segment symbols in the case they don't exist.
For example, with the `pipefifo_fops` we can try loading it in the eBPF program, and if it doesn't exist we can try matching some of the file operations pointers to expected functions symbols to verify it. You can see that the best use-case is with the struct, but it will just mean more complicated check if it is missing.